### ISSUE TITLE
[FEAT] A parent tree item should be expanded if there is an active/selected tree item within it

### ISSUE THREAD

**Issue by xelaint** [graycoreio/daffodil#2931]
# :bulb: Feature request

## Feature Name
A parent tree item should be expanded if there is an active/selected tree item within it

## Environment

<pre><code>
Daffodil version: X.Y.Z
Angular version: X.Y.Z 

 
For Tooling issues:
- Node version: XX  
- Platform:  

Others:

</code></pre>

**Comment by damienwebdev** [comment]
@xelaint is this permanently open? Or only open upon initial render?

**Comment by xelaint** [comment]
> @xelaint is this permanently open? Or only open upon initial render?

I think it should be permanent. If a user is on next.daff.io/docs/packages/cart/extension, and I refresh the page, the `@daffodil/cart` tree item should remain expanded so the user knows what page they're on and are able to find the related pages easily. Currently, if I navigate to Extension and refresh the page, the @daffodil/cart tree item is closed.

<img width="1061" alt="Screen Shot 2024-07-23 at 8 35 50 AM" src="https://github.com/user-attachments/assets/a081faa9-057a-4773-b9ba-4fdcd3d1a42f">

**Comment by damienwebdev** [comment]
Ok. If I'm on that page and I click:

![image](https://github.com/user-attachments/assets/8a6fe1de-2c90-4248-849d-354c7442b275)

What should happen?

Should the click do nothing?

**Comment by griest024** [comment]
I disagree. I think the feature should only cover what happens on initial render. I also think that it should scroll the active item into view.

**Comment by xelaint** [comment]
So if you're on the "Extension" page and decide to refresh to page, you wouldn't want to see that the Extension page is still the selected item in the sidebar?

**Comment by xelaint** [comment]
What I expect in a sidebar navigation where items are hidden:

1. User can collapse or expand a tree item, but the state resets upon refresh.
2. The selected tree item should be visible at all times.

Based on this UI, how does a user easily know where they are in the nav sidebar?

<img width="1060" alt="Screen Shot 2024-07-23 at 8 50 56 AM" src="https://github.com/user-attachments/assets/af721fee-9aad-471c-88cb-8ef300a7767d">


If a tree item isn't nested (i.e. I click on `@daffodil/authorizenet`), it's apparent to me that that's the active link. Why shouldn't that be the case for nested tree items?

**Comment by griest024** [comment]
> So if you're on the "Extension" page and decide to refresh to page, you wouldn't want to see that the Extension page is still the selected item in the sidebar?

refreshing the page is an "initial render"

the way I interpret "remain expanded" is that the accordion cannot be closed, even by user interaction.

**Comment by griest024** [comment]
my proposed UX:
- upon initial render (page load) the accordion that holds the active nested item (if the active one is nested) is expanded
- upon initial render the active nav item is scrolled into view

**Comment by xelaint** [comment]
Gotcha. I interpreted "initial render" as literally the first initial render and nothing after that. I think we're on the same page then.

**Comment by damienwebdev** [comment]
Fixed in https://github.com/graycoreio/daffodil/pull/4408 and merged as https://github.com/graycoreio/daffodil/commit/03cfe380faaf046fb7e6751c945f733cce5f0ba2
